examples/solids/qfunctions/FSInitial-MR1.h

```diff
@@ -0,0 +1,742 @@
+                                    ug[1][0][i],
+                                    ug[2][0][i]},
+                                   {ug[0][1][i],
+                                    ug[1][1][i],
+                                    ug[2][1][i]},
+                                   {ug[0][2][i],
+                                    ug[1][2][i],
+                                    ug[2][2][i]}
+                                  };
+    // -- Qdata
+    const CeedScalar dXdx[3][3] = {{q_data[1][i],
+                                    q_data[2][i],
+                                    q_data[3][i]},
+                                   {q_data[4][i],
+                                    q_data[5][i],
+                                    q_data[6][i]},
+                                   {q_data[7][i],
+                                    q_data[8][i],
+                                    q_data[9][i]}
+                                  };
+    // *INDENT-ON*
+
+    // Compute grad_u
+    //   dXdx = (dx/dX)^(-1)
+    // Apply dXdx to du = grad_u
+    CeedScalar grad_u[3][3];
+    for (int j = 0; j < 3; j++)     // Component
+      for (int k = 0; k < 3; k++) { // Derivative
+        grad_u[j][k] = 0;
+        for (int m = 0; m < 3; m++)
+          grad_u[j][k] += dXdx[m][k] * du[j][m];
+      }
+
+    // E - Green-Lagrange strain tensor
+    //     E = 1/2 (grad_u + grad_u^T + grad_u^T*grad_u)
+    const CeedInt indj[6] = {0, 1, 2, 1, 0, 0}, indk[6] = {0, 1, 2, 2, 2, 1};
+    CeedScalar E2work[6];
+    for (CeedInt m = 0; m < 6; m++) {
+      E2work[m] = grad_u[indj[m]][indk[m]] + grad_u[indk[m]][indj[m]];
+      for (CeedInt n = 0; n < 3; n++)
+        E2work[m] += grad_u[n][indj[m]]*grad_u[n][indk[m]];
+    }
+    // *INDENT-OFF*
+    CeedScalar E2[3][3] = {{E2work[0], E2work[5], E2work[4]},
+                           {E2work[5], E2work[1], E2work[3]},
+                           {E2work[4], E2work[3], E2work[2]}
+                          };
+    // *INDENT-ON*
+
+    // Displacement
+    diagnostic[0][i] = u[0][i];
+    diagnostic[1][i] = u[1][i];
+    diagnostic[2][i] = u[2][i];
+
+    // Pressure
+    const CeedScalar Jm1 = computeJM1(grad_u);
+    const CeedScalar logJ = log1p_series_shifted(Jm1);
+    // change logJ to Jm1 for (J-1) case
+    diagnostic[3][i] = -lambda*logJ;
+
+    // Stress tensor invariants
+    diagnostic[4][i] = (E2[0][0] + E2[1][1] + E2[2][2]) / 2.;
+    diagnostic[5][i] = 0.;
+    for (CeedInt j = 0; j < 3; j++)
+      for (CeedInt m = 0; m < 3; m++)
+        diagnostic[5][i] += E2[j][m] * E2[m][j] / 4.;
+    diagnostic[6][i] = Jm1 + 1;
+
+    // C : right Cauchy-Green tensor
+    // C = I + 2E
+    // *INDENT-OFF*
+    const CeedScalar C[3][3] = {{1 + E2[0][0], E2[0][1], E2[0][2]},
+                                {E2[0][1], 1 + E2[1][1], E2[1][2]},
+                                {E2[0][2], E2[1][2], 1 + E2[2][2]}
+                               };
+    // *INDENT-ON*
+    // compute CC = C*C = C^2
```

Comment:
Minor - first character in the comment line for Solids tends to be a capital letter